lumicks/pylake/group.py

```diff
@@ -21,3 +22,25 @@ def __getitem__(self, item):
         else:
             cls = channel_class(thing)
             return cls.from_dataset(thing)
+
+    def keys(self):
+        """Return group names at this level"""
+        return list(self.h5.keys())
```

Comment:
I'm a bit torn about this. On one hand, it makes it more convenient for users because they can see the items right away in a notebook cell. On the other hand, it goes against Python conventions and it can cause problems in case there are lots of keys. Usually, the returned keys are just a view which can be iterated without using memory for all the elements. When you call `list()` on it, it actually iterates over all the elements and puts them in a list which can be bad if you have a lot of them.

The downside is that users need to call `list(group.keys())` in order to see everything in a notebook, but that's the same thing they need to do if they were using `h5py` directly or anything that adopts a `dict`-like interface in Python.

If the goal is specifically to make it easier for users to see the thing, the better approach if what you're doing with `__repr__` below. Note that notebooks also support rich representations, e.g `_repr_html_`: https://ipython.readthedocs.io/en/stable/config/integrating.html#rich-display

Reply:
If I'm not mistaken (and in python, I'm often is)  this should return an object that is view on the group. Couldn't we implement _repr_ (or _repr_html_) on that view object that display all the possible keys?

Reply:
It may have been lost in my long comment above, but that is what I was trying to suggest :)

Reply:
Fair enough.

Actually, considering that list just iterates, this thing has an iterator, and you could just construct a list with list(group) now, I don't see a need for keys anymore and removed it.